includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -202,6 +227,35 @@ public function prepare_items() {
 		);
 	}
 
+	/**
+	 * Generate values needed for users table headers totals.
+	 * Values: $total_courses_started, $total_active_courses, $total_average_grade.
+	 *
+	 * @since  4.2.0
+	 */
+	private function set_users_table_header_total_values() {
+
+		// Get average grade for all the lessons graded for all the courses.
+		$this->total_average_grade = Sensei_Grading::get_graded_lessons_average_grade();
+
+		// Get the number of the courses that users have started.
+		$course_args_started   = array(
+			'type'   => 'sensei_course_status',
+			'status' => 'any',
+		);
+		$total_courses_started = Sensei_Utils::sensei_check_for_activity( $course_args_started );
```

Comment:
I'm not too sure we need to worry much about storing the result in a variable for this and course completed. `sensei_check_for_activity` calls `get_comments` under the hood, which I believe has some built-in caching. But I could be totally wrong on that. 😅  @m1r0 Any thoughts here?

The backstory to why this was added in the first place is [here](https://github.com/Automattic/sensei/pull/4904#discussion_r822536579).

Reply:
I totally missed that we want Students total: 
Here is the [change](https://github.com/Automattic/sensei/pull/4865/files#diff-6ce1dc075c4832d5bf050e621bcfecbab91d97504e8bc565f4f476b1ea2f654dR78):

Reply:
Fixed for the reports: 
| UI | Report |
| -- | -- |
| <img width="1340" alt="image" src="https://user-images.githubusercontent.com/7208249/157672936-4325edc2-57a9-450b-a167-a4b4c6e6ea23.png"> | <img width="931" alt="image" src="https://user-images.githubusercontent.com/7208249/157673034-7616a61b-f1b4-437d-9fc3-95e3f87f9838.png"> |

Reply:
> `sensei_check_for_activity` calls `get_comments` under the hood, which I believe has some built-in caching. But I could be totally wrong on that.

You are not wrong! `get_comments` is caching all results based on the provided query params. It uses memory cache which works on the current request only. So as long as the query params are the same the cache will be used.

Reply:
@donnapep  @m1r0 

So I removed the caching and storing in class variables altogether. Donna's PR will fix storing the value of the columns  so we can be sure nothing will be called 3 times in a single request.
Here is the [commit](https://github.com/Automattic/sensei/pull/4865/commits/3a1b373e0b91f02cb744f6861e008c30a7fb62a3)